File: example/src/main/java/com/dyhdyh/adapter/example/adapters/TextAdapter.java
package com.dyhdyh.adapter.example.adapters;

import android.view.ViewGroup;
import android.widget.TextView;

import com.dyhdyh.adapter.example.R;
import com.dyhdyh.adapters.BaseRecyclerAdapter;
import com.dyhdyh.adapters.holder.BaseViewHolder;

import java.util.List;

public class TextAdapter extends BaseRecyclerAdapter<String, TextAdapter.ItemViewHolder> {

    public TextAdapter(List<String> data) {
        super(data);
    }

    @Override
    public void onBindViewHolder(ItemViewHolder holder, int position, String item) {
        holder.tv.setText(item);
    }

    @Override
    public ItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ItemViewHolder(R.layout.item_text, parent);
    }

    protected static class ItemViewHolder extends BaseViewHolder {
        private TextView tv;

        public ItemViewHolder(int layoutId, ViewGroup parent) {
            super(layoutId, parent);
            tv = (TextView) itemView.findViewById(R.id.tv);
        }
    }

}

File: app/src/main/java/in/xiandan/adapters/example/adapters/TextAdapter.kt
package `in`.xiandan.adapters.example.adapters

import `in`.xiandan.adapters.BaseRecyclerAdapter
import `in`.xiandan.adapters.BaseViewHolder
import `in`.xiandan.adapters.example.R
import android.view.ViewGroup
import kotlinx.android.synthetic.main.item_text.view.*

/**
 * Created by xiandanin on 2021/1/12 15:07
 */
class TextAdapter(data:List<String>) : BaseRecyclerAdapter<String, BaseViewHolder>(data) {
    override fun onCreateViewHolderFormLayoutId(): Int {
        return R.layout.item_text
    }
    override fun onBindViewHolder(holder: BaseViewHolder, position: Int, item: String?) {
        holder.itemView.tv.setText(item)
    }



}